# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
# Commit: ef817155231f
# Full Hash: ef817155231f6fed522ce793cbc82b7be019de56
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Constify conduit getters. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D122244
# ==============================================================================

diff -r 310fdf39b051 -r ef817155231f dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
--- a/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Sun Apr 18 21:25:11 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Tue Aug 10 14:58:34 2021 +0200
@@ -201,7 +201,7 @@
    * Note: this is an ordered list and {a,b,c} != {b,a,c}
    */
   virtual bool SetLocalSSRCs(const Ssrcs& aSSRCs, const Ssrcs& aRtxSSRCs) = 0;
-  virtual Ssrcs GetLocalSSRCs() = 0;
+  virtual Ssrcs GetLocalSSRCs() const = 0;
 
   /**
    * Adds negotiated RTP header extensions to the the conduit. Unknown
@@ -217,7 +217,7 @@
       MediaSessionConduitLocalDirection aDirection,
       const RtpExtList& aExtensions) = 0;
 
-  virtual bool GetRemoteSSRC(Ssrc* ssrc) = 0;
+  virtual bool GetRemoteSSRC(Ssrc* ssrc) const = 0;
   virtual bool SetRemoteSSRC(Ssrc ssrc, Ssrc rtxSsrc) = 0;
   virtual bool UnsetRemoteSSRC(Ssrc ssrc) = 0;
   virtual bool SetLocalCNAME(const char* cname) = 0;
@@ -226,7 +226,7 @@
 
   virtual void SetSyncGroup(const std::string& group) = 0;
 
-  virtual bool HasCodecPluginID(uint64_t aPluginID) = 0;
+  virtual bool HasCodecPluginID(uint64_t aPluginID) const = 0;
 
   virtual void DeliverPacket(rtc::CopyOnWriteBuffer packet,
                              PacketType type) = 0;